src/client/opamSolution.ml

```diff
@@ -138,6 +138,36 @@ let atom_of_name name =
   name, None
 *)
 
+let did_you_mean st atoms = 
+  let open OpamPackage.Set.Op in 
+  let packages =
+    OpamFormula.packages_of_atoms ~disj:false
+      (st.packages ++ st.installed) atoms
```

Comment:
`st.packages ++ st.installed` is used twice in the function. You can create it once instead.
Also i don't actually remember off-the-top-of-my-head but have you checked that `st.packages` doesn't already contain `st.installed`?